src/core/agent/services/connectionService.ts

```diff
@@ -66,43 +67,60 @@ class ConnectionService extends AgentService {
     this.props.eventEmitter.on(EventTypes.ConnectionStateChanged, callback);
   }
 
+  onResolveOobi() {
+    this.props.eventEmitter.on(
+      EventTypes.ConnectionStateChanged,
+      (event: ConnectionStateChangedEvent) => {
+        if (event.payload.url) {
+          this.resolveOobi(event.payload.url);
+        }
+      }
+    );
+  }
+
+  onConnectionRemoved() {
+    this.props.eventEmitter.on(
+      EventTypes.ConnectionRemoved,
+      (data: ConnectionRemovedEvent) =>
+        this.deleteConnectionById(data.payload.connectionId!)
+    );
+  }
+
   @OnlineOnly
   async connectByOobiUrl(url: string): Promise<OobiScan> {
     const multiSigInvite = url.includes("groupId");
+    let connectionId = new URL(url).pathname
+      .split("/oobi/")
+      .pop()!
+      .split("/")[0];
 
-    // @TODO - foconnor: We shouldn't emit this if it's a multiSigInvite, but the routing will break if we don't.
-    // To fix once we handle errors for the scanner in general.
-    this.props.eventEmitter.emit<ConnectionStateChangedEvent>({
-      type: EventTypes.ConnectionStateChanged,
-      payload: {
-        isMultiSigInvite: multiSigInvite,
-        connectionId: undefined,
-        status: ConnectionStatus.PENDING,
-      },
-    });
+    const alias = new URL(url).searchParams.get("name") ?? new Salter({}).qb64;
 
-    const operation = await this.resolveOobi(url, multiSigInvite);
-    const connectionId =
-      operation.done && operation.response
-        ? operation.response.i
-        : new URL(url).pathname.split("/oobi/").pop()?.split("/")[0];
     const connectionMetadata: any = {
-      alias: operation.alias,
+      alias,
       oobi: url,
-      pending: !operation.done,
+      pending: false,
     };
     const groupId = new URL(url).searchParams.get("groupId") ?? "";
-    const connectionDate = operation.response?.dt ?? new Date();
+    const connectionDate = new Date().toISOString();
     const connection = {
       id: connectionId,
       connectionDate,
-      oobi: operation.metadata.oobi,
+      oobi: url,
       status: ConnectionStatus.CONFIRMED,
-      label: operation.alias,
+      label: alias,
       groupId,
     };
 
     if (multiSigInvite) {
+      const operation = await this.resolveOobi(url, multiSigInvite);
+
+      if (operation.done && operation.response) {
+        connectionId = operation.response.i;
+        connectionMetadata.pending = !operation.done;
+      }
```

Comment:
We know it's pending. We should just be able to do `connectionMetadata.pending = false;` without the if statement. We don't need to change the connectionId, we set it correctly already.

Reply:
You still need to add `connectionMetadata.pending = false;`...